test/e2e/util/util.go

```diff
@@ -1320,6 +1390,173 @@ func EnsureGuestWebhooksValidated(t *testing.T, ctx context.Context, guestClient
 	})
 }
 
+func EnsureKubeAPIDNSName(t *testing.T, ctx context.Context, mgmtClient crclient.Client, hc *hyperv1.HostedCluster) {
+	//AtLeast(t, Version419)
+	var (
+		hcKASCustomKubeconfigSecretName string
+		customApiServerHost             = "localhost"
```

Comment:
can we validate this with a real domain (that might just resolve to 127.0.0.1) and passing configuration.apisever.certs?

Reply:
yep, the plan I had is to add new E2E covering this onces this got merged. That works for you? (We need to got this merged before the feature freeze)

Reply:
without that test, there's zero evidence this works as it will be consumed and there's zero guarantees it won't be broken. We need that in place before this change ships into any publicly consumable artifact.